app/models/train.rb

```diff
@@ -8,18 +8,24 @@ class Train < ApplicationRecord
   has_many :tickets
   has_many :carriages
 
-  def train_carriages
-    @train_carriages ||= {}
-  end
-
   def train_carriages_info
-    types = carriages.map(&:carriage_type).uniq
-    types.each do |type|
-      carriages_by_type = carriages.select { |carriage| carriage.carriage_type == type }
-      train_carriages[type] = { top_seats_count: carriages_by_type.sum(&:top_seats_count),
-                                lower_seats_count: carriages_by_type.sum(&:lower_seats_count),
-                                size: carriages_by_type.size }
+    train_carriages = {}
+    Carriage.carriage_types.each do |type|
```

Comment:
Looks like this functionality preparing data for view please move this functionality to the helper.